### DISCUSSION TITLE
Replace a Block Reference with its source text content

### DISCUSSION THREAD

**Discussion by GitMurf** [SilentVoid13/Templater#166] [Templates Showcase]
https://gist.github.com/GitMurf/c34ab5aedb2161ac26cc8076a116000a

Replace a block ref in the current line with the actual text from the block ref. It replicates Roam's feature where you right-click a block ref and choose "Replace as text".

QUICK DEMO: https://user-images.githubusercontent.com/64155612/115739741-05c70f00-a343-11eb-970c-c944783414a8.mp4

**Comment by ab-ll** [comment]
I like the idea of this - is it possible to dynamically select the file this goes into?  Specifically I'd like to be able to select text and have it automatically go into the daily notes for that day, into the same heading if it exists with a timestamp next to it.  So, today's would go to today - and tomorrow's would go to tomorrow.  I use the review plugin to put references to a file I'm working on into the review section of the daily notes, but this would be better if it was text (potentially with a link to the file working on)

**Reply by GitMurf** [reply]
I'm not quite following you on this. Can you please record a video or give some screenshots of what you are looking for? Thanks.

**Comment by trungng2006** [comment]
Thank you for this gist. 
Could you also kindly also implement this code for headings ref also?
